ydb/core/blobstorage/dsproxy/request_history.h

```diff
@@ -0,0 +1,206 @@
+        TVGetEntry(TInstant startTime, ui8 partIdx, ui32 queryCount, ui8 orderNumber)
+            : TVRequestEntry(startTime, partIdx, queryCount,orderNumber)
+        {}
+
+        void Output(IOutputStream& str, const TIntrusivePtr<TBlobStorageGroupInfo>& info,
+                const TLogoBlobID* blobId) const {
+            TVRequestEntry::Output(str, "TEvVGet", info, blobId);
+        }
+    };
+
+    struct TVPutResultEntry : public TVResponseEntry {
+        TVPutResultEntry(TInstant startTime, ui8 orderNumber, NKikimrProto::EReplyStatus status, const TString& errorReason)
+            : TVResponseEntry(startTime, orderNumber, status, errorReason)
+        {}
+
+        void Output(IOutputStream& str, const TIntrusivePtr<TBlobStorageGroupInfo>& info,
+                const TLogoBlobID* blobId) const {
+            Y_UNUSED(blobId);
+            TVResponseEntry::Output(str, "TEvVPutResult", info);
+        }
+    };
+
+    struct TVGetResultEntry : public TVResponseEntry {
+        TVGetResultEntry(TInstant startTime, ui8 orderNumber, NKikimrProto::EReplyStatus status, const TString& errorReason)
+            : TVResponseEntry(startTime, orderNumber, status, errorReason)
+        {}
+
+        void Output(IOutputStream& str, const TIntrusivePtr<TBlobStorageGroupInfo>& info,
+                const TLogoBlobID* blobId) const {
+            Y_UNUSED(blobId);
+            TVResponseEntry::Output(str, "TEvVGetResult", info);
+        }
+    };
+
+    struct TAccelerationEntry : public TBaseEntry {
+        TAccelerationEntry(TInstant startTime)
+            : TBaseEntry(startTime)
+        {}
+
+        void Output(IOutputStream& str, const char* typeName) const {
+            str << typeName << "{ TimestampMs# " << Timestamp.MilliSeconds() << " }";
+        }
+    };
+
+    struct TGetAccelerationEntry : public TAccelerationEntry {
+        TGetAccelerationEntry(TInstant startTime)
+            : TAccelerationEntry(startTime)
+        {}
+        void Output(IOutputStream& str, const TIntrusivePtr<TBlobStorageGroupInfo>& info,
+                const TLogoBlobID* blobId) const {
+            Y_UNUSED(info);
+            Y_UNUSED(blobId);
+            TAccelerationEntry::Output(str, "GetAcceleration");
+        }
+    };
+
+    struct TPutAccelerationEntry : public TAccelerationEntry {
+        TPutAccelerationEntry(TInstant startTime)
+            : TAccelerationEntry(startTime)
+        {}
+
+        void Output(IOutputStream& str, const TIntrusivePtr<TBlobStorageGroupInfo>& info,
+                const TLogoBlobID* blobId) const {
+            Y_UNUSED(info);
+            Y_UNUSED(blobId);
+            TAccelerationEntry::Output(str, "PutAcceleration");
+        }
+    };
+
+
+    using TEntry = std::variant<TVPutEntry, TVGetEntry, TVPutResultEntry, TVGetResultEntry, TPutAccelerationEntry,
+            TGetAccelerationEntry>;
+
+private:
+
+    const TIntrusivePtr<TBlobStorageGroupInfo> Info;
+    TInstant StartTime;
```

Comment:
Maybe THPTimer? Or TMonotonic, if you want this in tests. But I don't think there is too much to test in here.